controllers/slice/namespaces.go

```diff
@@ -719,8 +720,23 @@ func (r *SliceReconciler) labelAppNamespaces(ctx context.Context, cfgAppNsList [
 		namespace := &corev1.Namespace{}
 		err := r.Get(ctx, types.NamespacedName{Name: cfgAppNs}, namespace)
 		if err != nil {
-			log.Error(err, "Failed to find namespace", "namespace", cfgAppNs)
-			continue
+			// If the namespace doesn't exist on specified cluster then first create it
+			if errors.IsNotFound(err) {
+				log.Info("Namespace is not found. Creating namespace.", "namespace", cfgAppNs)
+				namespace = &corev1.Namespace{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: cfgAppNs,
+					},
+				}
+				if err := r.Create(ctx, namespace); err != nil {
+					log.Error(err, "Failed to create namespace", "namespace", cfgAppNs)
+					continue
```

Comment:
is there any reason we are continuing in case of error? and not returning error?

Reply:
Suppose there is a scenario where we create a slice CR with app ns array in which we mention an invalid NS name followed by some valid ones. In such case I thought of skipping the invalid one while creating(and labeling) rest of them. That's why I used `continue` instead of `return`. Let me know what you think about this approach?